lib/chef/resource/file.rb

```diff
@@ -50,80 +48,15 @@ class File < Chef::Resource
       default_action :create
       allowed_actions :create, :delete, :touch, :create_if_missing
 
-      def initialize(name, run_context=nil)
-        super
-        @path = name
-        @backup = 5
-        @atomic_update = Chef::Config[:file_atomic_update]
-        @force_unlink = false
-        @manage_symlink_source = nil
-        @diff = nil
-        @verifications = []
-      end
-
-      def content(arg=nil)
-        set_or_return(
-          :content,
-          arg,
-          :kind_of => String
-        )
-      end
-
-      def backup(arg=nil)
-        set_or_return(
-          :backup,
-          arg,
-          :kind_of => [ Integer, FalseClass ]
-        )
-      end
-
-      def checksum(arg=nil)
-        set_or_return(
-          :checksum,
-          arg,
-          :regex => /^[a-zA-Z0-9]{64}$/
-        )
-      end
-
-      def path(arg=nil)
-        set_or_return(
-          :path,
-          arg,
-          :kind_of => String
-        )
-      end
-
-      def diff(arg=nil)
-        set_or_return(
-          :diff,
-          arg,
-          :kind_of => String
-        )
-      end
-
-      def atomic_update(arg=nil)
-        set_or_return(
-          :atomic_update,
-          arg,
-          :kind_of => [ TrueClass, FalseClass ]
-        )
-      end
-
-      def force_unlink(arg=nil)
-        set_or_return(
-          :force_unlink,
-          arg,
-          :kind_of => [ TrueClass, FalseClass ]
-        )
-      end
-
-      def manage_symlink_source(arg=nil)
-        set_or_return(
-          :manage_symlink_source,
-          arg,
-          :kind_of => [ TrueClass, FalseClass ]
-        )
-      end
+      property :content, [ String, nil ], desired_state: false
```

Comment:
This is going to apply to all the similar pull reqs, but we should probably be more explicit that we are okay with not using parens here. The usual style for the rest of the Chef code is to use parens in most cases, and these classes are complex enough that I don't think anyone would mistake them for a pseudo-declarative thang. The plus side is it would be nice to have the Chef code look style-wise like we tell people to write LWRPs.